### DISCUSSION TITLE
Local development vs Real deployment

### DISCUSSION THREAD

**Discussion by devmousa** [Cyfrin/foundry-full-course-cu#707] [General]
I've a *question*:
I've completed foundry simple storage and will complete foundry fund me very soon
but now, I'm wondering
when we will deploy our contract to a real testnet or a real mainnet, we won't use script folder anymore, right ?
is ```script``` folder has the same purpose as the ```test``` folder ? (I mean, run and test the app locally)
as far as I understand, we use script folder to make everything works programmatically and don't repeat the same operation with the terminal
so I understand that the both script and test folders have locally purposes only (to make sure that our contracts are working very well before deploy them to the real world)

are my thoughts are correct ?
are test and script folders for the locally development only ?
or they have to run also in real world ? (I don't know how 😅)

thank you all in advance and happy coding ;)

**Comment by theirrationalone** [comment]
We also use scripts to deploy intended contracts on testnets as well as on mainnet... Tests can also be performed on mainnet but that would cost too much... 

Unlikely, as you said...but if we have enough testnets eths then we also can perform tests on testnets, a consciously aware dev should test his/her contracts on testnets as well as on forknets, atleast once most likely just after the end of development & local testing.😉

Feel free to correct me, If i went wrong somewhere, somehow. 😅

Thank You 🙏
